src/components/CustomerDetails/CustomerDetails.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import Checkbox from "../Checkbox/Checkbox";
+import styles from "./CustomerDetails.module.scss";
+import { customerDetails } from "../../constants/customerDetailsMapping";
+import { CustomerDetailsProps } from "../../types/customerDetailsPropsTypes";
+
+const CustomerDetails: React.FC<CustomerDetailsProps> = ({
+  customer,
+  isChecked,
+  isPartiallyChecked,
+  onCheckboxChange,
+}) => {
+  return (
+    <>
+      <Checkbox
+        isChecked={isChecked}
+        isPartiallyChecked={isPartiallyChecked}
+        onChange={onCheckboxChange}
+      />
+      <div className={styles.customerName}>{customer.name}</div>
+      <div className={styles.customerTable}>
+        <div className={styles.customerHeader}>
+          {customerDetails.map((detail) => (
+            <span key={detail.key}>{detail.label}</span>
+          ))}
+        </div>
+        <div className={styles.customerRow}>
+          {customerDetails.map((detail) => (
+            <span key={detail.key}>
+              {typeof customer[detail.key] === "string" ||
```

Comment:
any specific reason to check if String or Number?

Reply:
I used this type checking to make sure only strings or numbers are rendered, but since it's certain that customer[detail.key] will contain strings / numbers, therefore no need of type checking. I'll be removing that.